deepchem/data/data_loader.py

```diff
@@ -205,7 +205,19 @@ def shard_generator():
         time1 = time.time()
         X, valid_inds = self._featurize_shard(shard)
         ids = shard[self.id_field].values
-        ids = ids[valid_inds]
+        #  special case when we deal with  dataset of molecular fragments:
```

Comment:
As a general design comment, `DataLoader` is a generic class so we shouldn't have a special case for `ConvMolFeaturizer`. Can't this be done within `ConvMolFeaturizer.featurize()` itself?

Reply:
I'll  try avoiding changes in **data_loader.py** copletely, but then I will probably need to make some adjustment in **deepcheem/feat/base_classes.py**, namely, in `MolecularFeaturizer` class. Is it acceptable at all in general?

Reply:
I think you can just override the implementations directly in `ConvMolFeaturizer`. For example, just override the `featurize()` or other methods implemented in `MolecularFeaturizer` directly in `ConvMolFeaturizer`.